lib/babe/build.go

```diff
@@ -180,42 +177,40 @@ func (b *Service) buildBlockBabeHeader(slot Slot) (*types.BabeHeader, error) {
 // buildBlockExtrinsics applies extrinsics to the block. it returns an array of included extrinsics.
 // for each extrinsic in queue, add it to the block, until the slot ends or the block is full.
 // if any extrinsic fails, it returns an empty array and an error.
-func (b *Service) buildBlockExtrinsics(slot Slot) ([]*transaction.ValidTransaction, error) {
+func (b *Service) buildBlockExtrinsics(slot Slot) []*transaction.ValidTransaction {
 	next := b.nextReadyExtrinsic()
 	included := []*transaction.ValidTransaction{}
 
 	for !hasSlotEnded(slot) && next != nil {
 		b.logger.Trace("build block", "applying extrinsic", next)
+		t := b.transactionState.Pop()
 		ret, err := b.rt.ApplyExtrinsic(next)
+
 		if err != nil {
-			return nil, err
+			b.logger.Trace("apply extrinsic failed:", err, "extrinsic", next)
```

Comment:
can you add an `"error"` in from of the `err`? the logger expects an even number of arguments, after the initial info string

Reply:
@noot 

Do you mean to replace the line as: 
```			b.logger.Trace("apply extrinsic failed:", "error", err, "extrinsic", next)
```

Reply:
yeah exactly, also change to `Warn` instead of `Trace`?